src/test/datascience/notebook/helper.ts

```diff
@@ -251,13 +251,11 @@ export async function createTemporaryNotebook(
     const data: nbformat.INotebookContent = {
         cells,
         metadata: {
-            orig_nbformat: 4
+            orig_nbformat: 4,
+            kernelspec: kernelSpec
         },
         nbformat: 4,
-        nbformat_minor: 2,
-        kernel: {
```

Comment:
This is an invalid INotebookContent and nbconvert wasn't liking the <kernel> entry.